test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,288 @@
+        ]),
+     ]),
+    nt.nested_tensor([
+        nt.nested_tensor([
+            torch.tensor([[5, 6],
+                          [7, 8]])
+        ]),
+    ])
+])
+
+original_nt2 = nt.nested_tensor([
+    nt.nested_tensor([
+        nt.nested_tensor([
+            torch.tensor([1, 2])
+        ]),
+    ]),
+    nt.nested_tensor([
+        nt.nested_tensor([
+            torch.tensor([3, 4]),
+            torch.tensor([5, 6])
+        ])
+    ])
+])
+        
+expected_nt0 = torch.tensor([[[[[1, 2],
+                                [3, 4]]]],
+                             [[[[5, 6],
+                                [7, 8]]]]])
+
+expected_nt1 = nt.nested_tensor([
+    torch.tensor([[[[1, 2],
+                    [3, 4]]]]),
+    torch.tensor([[[[5, 6],
+                    [7, 8]]]]),
+])
+
+expected_nt2 = nt.nested_tensor([
+    nt.nested_tensor([
+        torch.tensor([[[1, 2],
+                       [3, 4]]]),
+    ]),
+    nt.nested_tensor([
+        torch.tensor([[[5, 6],
+                      [7, 8]]]),
+    ]),
+])
+
 class TestTensorMask(TestCase):
-    # TODO: test comparison operator fails if nested_size doesn't agree.
-
-    def test_simple(self):
-        values = [torch.rand(i) for i in range(10)]
-        random.shuffle(values)
-        nt = nestedtensor.nested_tensor(values)
-        tensor, mask = nt.to_tensor_mask()
-
-    def test_nested(self):
-        nt = utils.gen_nested_tensor(2, 2, 2)
-        tensor, mask = nt.to_tensor_mask()
-
-    def test_tensor_mask(self):
-        nt = utils.gen_nested_tensor(2, 2, 2, size_low=1, size_high=2)
-        tensor, mask = nt.to_tensor_mask()
-        nt1 = nestedtensor.nested_tensor_from_tensor_mask(
-            tensor, mask, nested_dim=nt.nested_dim())
-        self.assertEqual(nt, nt1)
-        nt2 = nestedtensor.nested_tensor_from_tensor_mask(
-            tensor, mask)
-        self.assertEqual(nt, nt2)
-
-    def test_tensor_mask_lowdim(self):
-        values = [torch.rand(1, 2) for i in range(10)]
-        values = [values[1:i] for i in range(2, 10)]
-        nt = nestedtensor.nested_tensor(values)
-        tensor, mask = nt.to_tensor_mask()
-        self.assertTrue([len(value) for value in values] == list(mask.sum(-1)))
+    def test_gen_nested_tensor(self):
+        nt1 = utils.gen_nested_tensor(seed=1, nested_dim=1, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(nt.is_nested_tensor(nt1))
+        self.assertEqual(nt1.nested_dim(), 1)
+        self.assertEqual(nt1.size(), (2, 2))
+        self.assertEqual(nt1[0].dim(), 1)
+        self.assertEqual(nt1[0].size(), torch.Size([2]))
+
+        nt1 = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(nt.is_nested_tensor(nt1))
+        self.assertEqual(nt1.nested_dim(), 2)
+        self.assertEqual(nt1.size(), (2, 2, 2))
+        self.assertEqual(nt1[0].dim(), 2)
+        self.assertEqual(nt1.nested_size(), nt.NestedSize([[[2],[2]], [[2],[2]]]))
```

Comment:
After https://github.com/pytorch/nestedtensor/commit/02ef5f895a8c21f5f304d625c320b60c231df984 you need to use ```from utils import nested_size_to_list``` instead of the constructor or introduce the constructor.